src/__test__/components/data-management/ProjectMenu.test.jsx

```diff
@@ -91,4 +117,29 @@ describe('ProjectMenu', () => {
 
     await waitFor(() => expect(screen.getByText('Upload')).toBeInTheDocument());
   });
+
+  it('Clicking on Share button opens share experiment modal', async () => {
+    await act(async () => {
+      render(
+        <Provider store={storeState}>
+          {projectMenuFactory()}
+        </Provider>,
+      );
+    });
+
+    const shareButton = screen.getByText('Share').closest('button');
+
+    await act(async () => {
+      userEvent.click(shareButton);
+    });
+
+    await waitFor(() => expect(screen.getByText('Share with collaborators')).toBeInTheDocument());
```

Comment:
Can you check if you need this `waitFor`? I think by wrapping the `act` above, we should already give time for the UI to finish its rendering.